.github/unittest.sh

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+echo '::group::Prepare conda'
+CONDA_PATH=$(which conda)
+eval "$(${CONDA_PATH} shell.bash hook)"
+# The `setuptools` package installed through `conda` includes a patch that errors if something is installed
+# through `setuptools` while the `CONDA_BUILD` environment variable is set.
+# https://github.com/AnacondaRecipes/setuptools-feedstock/blob/f5d8d256810ce28fc0cf34170bc34e06d3754041/recipe/patches/0002-disable-downloads-inside-conda-build.patch
+# (Although we are not using the `-c conda-forge` channel, the patch is equivalent but not public for
+# `setuptools` from the `-c defaults` channel)
+# Since we aren't using `conda build` here, we unset it to avoid installation problems later
+# TODO: investigate where `CONDA_BUILD` is set and maybe fix unset it there
+unset CONDA_BUILD
+echo '::endgroup::'
+
+echo '::group::Set PyTorch conda channel'
+# TODO: Can we maybe have this as environment variable in the job template? For example, `IS_RELEASE`.
+if [[ (${GITHUB_EVENT_NAME} = 'pull_request' && (${GITHUB_BASE_REF} = 'release'*)) || (${GITHUB_REF} = 'refs/heads/release'*) ]]; then
+  POSTFIX=test
+else
+  POSTFIX=nightly
+fi
+PYTORCH_CHANNEL=pytorch-"${POSTFIX}"
+echo "${PYTORCH_CHANNEL}"
+echo '::endgroup::'
+
+echo '::group::Set PyTorch GPU mutex'
+case $GPU_ARCH_TYPE in
```

Comment:
Not needed for macOS since we don't have GPU workflows. This is in preparation for the Linux GPU jobs. Will happen a few more times below.

Reply:
See #7189.